src/terms/hubbard.jl

```diff
@@ -107,51 +107,68 @@ Hubbard energy, following the Dudarev et al. (1998) rotationally invariant forma
+    end
+    function Hubbard(params::Dict)
+        Hubbard(collect(keys(params)), collect(values(params)))
+    end
+    function Hubbard(params::Pair)
+        Hubbard([params[1]], [params[2]])
     end
 end
 function (hubbard::Hubbard)(basis::AbstractBasis)
-    manifold = resolve_hubbard_manifold(hubbard.manifold, basis.model)
-
+    manifolds = [resolve_hubbard_manifold(manifold, basis.model) for manifold in hubbard.manifolds]
     projs, labels = atomic_orbital_projectors(basis)
-    labels, projectors_matrix = extract_manifold(manifold, projs, labels)
-    TermHubbard(manifold, hubbard.U, projectors_matrix, labels)
+    manifold_labels, projs_matrices = [], []
+    for manifold in manifolds
+        push!(manifold_labels, extract_manifold(manifold, projs, labels).manifold_labels)
+        push!(projs_matrices, extract_manifold(manifold, projs, labels).manifold_projectors)
+    end
+    TermHubbard(manifolds, hubbard.U, projs_matrices, manifold_labels)
 end
 
 struct TermHubbard{T, PT, L} <: Term
-    manifold::ResolvedOrbitalManifold
-    U::T   # U value
-    P::PT  # projectors
-    labels::L
+    manifolds::Vector{ResolvedOrbitalManifold}
+    U::Vector{T}  # U values
+    P::Vector{PT} # projectors per manifold
+    labels::Vector{L}
 end
 
-@timing "ene_ops: hubbard" function ene_ops(term::TermHubbard,
-                                            basis::PlaneWaveBasis{T},
+@timing "ene_ops: hubbard" function ene_ops(term::TermHubbard, 
+                                            basis::PlaneWaveBasis{T}, 
                                             ψ, occupation; hubbard_n=nothing,
                                             kwargs...) where {T}
     if isnothing(hubbard_n)
        return (; E=zero(T), ops=[NoopOperator(basis, kpt) for kpt in basis.kpoints])
     end
-    proj = term.P
+    # Concatenate projectors from all manifolds per k-point as well as U value for each projector
+    projs = [hcat((P_manifold[ik] for P_manifold in term.P)...) for ik in 1:length(term.P[1])]
+    U = [term.U[iman] for iman in 1:length(term.manifolds) for _ in 1:size(term.P[iman][1],2)]
 
     filled_occ = filled_occupation(basis.model)
-    natoms = length(term.manifold.iatoms)
     n_spin = basis.model.n_spin_components
-    nproj_atom = size(hubbard_n[1,1,1], 1) # This is the number of projectors per atom, namely 2l+1
-    # For the ops we have to reshape hubbard_n to match the NonlocalOperator structure, using a block diagonal form
-    nhub = [zeros(Complex{T}, nproj_atom*natoms, nproj_atom*natoms) for _ in 1:n_spin]
+    # For the ops we have to reshape hubbard_n to match the NonlocalOperator structure, 
+    # using a block diagonal form whit one block per atom per manifold
+    # 2l+1 is the number of projectors per atom, i.e. the dimension of the blocks
+    nhub_size = sum((2*manifold.l+1)*length(manifold.iatoms) for manifold in term.manifolds)
+    nhub = [zeros(Complex{T}, nhub_size, nhub_size) for _ in 1:n_spin]
```

Comment:
Can we maybe use `D_size` and `D` for the variables? To make it clear that this is all about building the diagonal matrix for the nonlocal operator.